packages/core/src/common/metrics.ts

```diff
@@ -255,72 +260,87 @@ export async function getHistoricalSyncProgress(metrics: MetricsService) {
     }
   }
 
-  /** Aggregate block metrics for different "types" of sources. */
-  const reduceBlockMetrics = (values: prometheus.MetricValue<"network">[]) =>
-    values.reduce<{
-      [network: string]: number;
-    }>((acc, cur) => {
-      const network = cur.labels.network as string;
-      acc[network] = cur.value;
-      return acc;
-    }, {});
-
-  const cachedBlocksMetric = await metrics.ponder_historical_cached_blocks
-    .get()
-    .then(({ values }) => reduceBlockMetrics(values));
-  const totalBlocksMetric = await metrics.ponder_historical_total_blocks
-    .get()
-    .then(({ values }) => reduceBlockMetrics(values));
-  const completedBlocksMetric = await metrics.ponder_historical_completed_blocks
-    .get()
-    .then(({ values }) => reduceBlockMetrics(values));
-
-  const networks = Object.entries(totalBlocksMetric).map(
-    ([network, totalBlocks]) => {
-      const cachedBlocks = cachedBlocksMetric[network];
-      const completedBlocks = completedBlocksMetric[network] ?? 0;
-
-      // If cachedBlocks is not set, setup is not complete.
-      if (cachedBlocks === undefined) {
-        return {
-          networkName: network,
-          totalBlocks,
-          completedBlocks,
-        };
+  const extractMetric = (
+    metric: prometheus.MetricObjectWithValues<
+      prometheus.MetricValue<"network">
+    >,
+    network: string,
+  ) => {
+    return metric.values.find((m) => m.labels.network === network)?.value;
+  };
+
+  const totalBlocksMetric = await metrics.ponder_historical_total_blocks.get();
```

Comment:
It's a pain that all of these have to be async when we know that there is no async `collect` function registered for any of our metrics. As a nerdsnipe, we could create a `SyncRegistry` subclass that enforces sync for everything.